docs/scos/dev/tutorials-and-howtos/advanced-tutorials/tutorial-howto-import-1m-of-products-and-make-it-searchable-and-shown-on-the-storefront.md

```diff
@@ -0,0 +1,70 @@
+
+This approach way more performant, but it creates several difficulties, namely:
+
+* Tracking errors becomes more difficult because DataImport now writes multiple entities to the database simultaneously. If the data contains an error, it's hard to determine which data set caused the problem (this issue should be addressed during data validation).
+* You will need to maintain a complex raw SQL query and modify it in case of changes to the data structure or import functionality.
+* The SQL query structure will likely differ depending on the selected RDBMS (PostgreSQL/MySQL).
+
+### Raw SQL queries and Common Table Expressions (CTE)
+
+In the paragraph above, we have learned that a raw SQL query provides better performance when importing large volumes of data and is the preferred method for storing information in a database. However, there are several approaches to consider, and we will discuss two of them:
+
+* Common Table Expressions (CTE)
+* UPSERT using INSERT with ON DUPLICATE KEY UPDATE
+
+#### Common Table Expressions (CTE)
+
+CTEs allow users to create additional statements that can be utilized within a larger query. These statements function as temporary tables that are only applicable to a single query. The auxiliary statements, which can take the form of SELECT, INSERT, UPDATE, or DELETE commands, are grouped together within a WITH clause that is then attached to a primary statement. The primary statement can also be a SELECT, INSERT, UPDATE, or DELETE command.
+
+By utilizing CTEs, users can simplify the process of writing and maintaining complex queries, resulting in increased readability. This is achieved by breaking down complicated queries into simple blocks that can be reused when needed, similar to how database views and derived tables work. As a result, the overall complexity of the query is reduced.
+
+As we can see, CTE is a very convenient tool, but how about its performance? It all depends on the RDBMS you are using, consider to check RDBMS documentation before using it, as there is a possibility that you will not get any performance improvements.
+
+#### UPSERT using INSERT with ON DUPLICATE KEY UPDATE
+
+This approach is easy to use, provides good and predictable results, but slightly complicates the final code of the importer.
+
+## Best Practices
+
+* Use bulk writes for your data importer, as it's dramatically increases performance of write operations and whole data import process. You can find exmaples of bulk writers in our boilerplate, e.g.: [ProductAbstractBulkPdoDataSetWriter.php](https://github.com/spryker-shop/suite/blob/35fdc84a67c4d6f01b235ff3101a6bbd137025b1/src/Pyz/Zed/DataImport/Business/Model/ProductAbstract/Writer/ProductAbstractBulkPdoDataSetWriter.php)
+* Avoid using synchronous commits in the data import code
```

Comment:
Could you elaborate a bit more here, what does that mean and why does it need to be avoided?